autopts/client.py

```diff
@@ -1373,15 +1373,22 @@ def sigint_handler(sig, frame):
             signal.signal(signal.SIGINT, sigint_handler)
 
             return self.main(args)
-        except BaseException as e:
-            if not isinstance(e, KeyboardInterrupt):   # Ctrl-C
-                if e.code != 0:
-                    # Exit with traceback
-                    logging.exception(e)
-
+        except KeyboardInterrupt:
+            # Handling Ctrl-C
+            raise
+        except SystemExit as e:
+            if e.code != 0:
+                logging.exception(e)
+            # Handling --help
+            raise
+        except (TimeoutError, BaseException) as e:
+            logging.exception(e)
+            # We have to propagate an exception from the simple client layer
+            # up to the bot layer, so it could handle its own cleanup.
+            raise
```

Comment:
I think you can put the `raise` in the `finally` clause, right?

Reply:
It seems that a lonely `raise` doesn't work in `finally`:
`RuntimeError: No active exception to reraise`